packages/client/components/ActivityLibrary/ScheduleMeetingButton.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {useEffect, useState} from 'react'
+import graphql from 'babel-plugin-relay/macro'
+import {useDialogState} from '../../ui/Dialog/useDialogState'
+import SecondaryButton from '../SecondaryButton'
+import GcalModal from '../../modules/userDashboard/components/GcalModal/GcalModal'
+import {CreateGcalEventInput} from '../../__generated__/StartRetrospectiveMutation.graphql'
+import GcalClientManager from '../../utils/GcalClientManager'
+import useAtmosphere from '../../hooks/useAtmosphere'
+import {useFragment} from 'react-relay'
+import {ScheduleMeetingButton_team$key} from '~/__generated__/ScheduleMeetingButton_team.graphql'
+import {ScheduleMeetingButton_viewer$key} from '~/__generated__/ScheduleMeetingButton_viewer.graphql'
+import {MenuMutationProps} from '../../hooks/useMutationProps'
+
+type Props = {
+  mutationProps: MenuMutationProps
+  handleStartActivity: (gcalInput?: CreateGcalEventInput) => void
+  teamRef: ScheduleMeetingButton_team$key
+  viewerRef: ScheduleMeetingButton_viewer$key
+}
+
+const ScheduleMeetingButton = (props: Props) => {
+  const {mutationProps, handleStartActivity, teamRef, viewerRef} = props
+  const {
+    isOpen: isScheduleDialogOpen,
+    open: openScheduleDialog,
+    close: closeScheduleDialog
+  } = useDialogState()
+  const atmosphere = useAtmosphere()
+  const [hasStartedGcalAuthTeamId, setHasStartedGcalAuthTeamId] = useState<null | string>(null)
+  const {submitting} = mutationProps
+
+  const viewer = useFragment(
+    graphql`
+      fragment ScheduleMeetingButton_viewer on User {
+        featureFlags {
+          gcal
+        }
+      }
+    `,
+    viewerRef
+  )
+  const hasGcalFlag = viewer.featureFlags.gcal
+
+  const team = useFragment(
+    graphql`
+      fragment ScheduleMeetingButton_team on Team {
+        id
+        teamMembers {
+          isSelf
+          integrations {
+            gcal {
+              auth {
+                id
+              }
+              cloudProvider {
```

Comment:
+1 not blocking yet, but could be an issue for PPMIs once we remove the feature flag. I did have the feature flag but no `cloudProvider` yet, so I saw the "schedule" button which just was a no-op. Maybe it would be cleaner to:
* on server side check if the user has the feature flag and only then return the `cloudProvider`, otherwise return `null
* on client don't check the feature flag at all but just the presence of `cloudProvider`

Reply:
Sounds good - I'm now only showing the "Schedule" button if the gcal flag & the `cloudProvider` exist. I've also implemented a `viewerTeamMember` field on the `Team` object so we only query one team member.